csvtools.nim

```diff
@@ -351,4 +351,10 @@ proc writeToCsv*[T](ts: openarray[T], path: string, separator = ',',
   var f = open(path, fmWrite)
   defer:
     f.close()
-  writeToCsv(ts, f, separator, quote, escape, quoteAlways, dateLayout)
\ No newline at end of file
+
+  if writeHeaders:
+      var headers = newSeq[string]()
+      for a,b in fieldPairs(T()):
+        headers.add a
+      f.write(headers.connect())
```

Comment:
Here you should pass all relevant parameters (separator, escape, etc)

Reply:
passed parameters